src/noit_message_decoder.c

```diff
@@ -397,7 +397,7 @@ noit_metric_tags_count(const char *str, size_t strlen) {
 
 const char *
 noit_metric_tags_parse_one(const char *tagnm, size_t tagnmlen,
-                           noit_metric_tag_t *output) {
+                           noit_metric_tag_t *output, mtev_boolean *toolong) {
   size_t colon_pos = 0;
   size_t cur_size = 0;
```

Comment:
might want to:

`*toolong = mtev_false;` 

at the top of this function due to return on line 422